src/d3a/models/strategy/area_agents/two_sided_pay_as_bid_engine.py

```diff
@@ -132,9 +132,16 @@ def event_bid_traded(self, *, bid_trade):
             if bid_trade.offer_bid_trade_info is not None:
                 # Adapt trade_offer_info received by the trade to include source market grid fees,
                 # which was skipped when accepting the bid during the trade operation.
+                trade_offer_info = TradeBidOfferInfo(
+                    original_bid_rate=bid_trade.offer_bid_trade_info.original_bid_rate,
+                    propagated_bid_rate=bid_trade.offer_bid_trade_info.propagated_bid_rate,
+                    original_offer_rate=bid_trade.offer_bid_trade_info.original_offer_rate,
+                    propagated_offer_rate=bid_trade.offer_bid_trade_info.propagated_offer_rate,
+                    trade_rate=bid_trade.offer_bid_trade_info.trade_rate
+                )
                 updated_trade_offer_info = \
                     self.markets.source.fee_class.propagate_original_offer_info_on_bid_trade(
-                        [None, None, *bid_trade.offer_bid_trade_info]
+                        trade_offer_info
```

Comment:
```suggestion
                        bid_trade.offer_bid_trade_info
```
Sorry I know I was the one suggested earlier to create [here](https://github.com/gridsingularity/d3a/pull/943/files#diff-4c2698dbd37d0660ae1460c62b57278e468c8d9674f2ed19b95a59b5d251ff64R135-R141) `TradeBidOfferInfo` but now that I'm looking into it again realised `bid_trade` already has an `offer_bid_trade_info` and recreation of `TradeBidOfferInfo` could simply be avoided in this case .

Reply:
It was failing with the list before, so it was a fix, not a suggestion I think

Reply:
Indeed it will fail with list but what I am proposing now is to replace `list` with `bid_trade.offer_bid_trade_info` which is already in the  `TradeBidOfferInfo ` format and a duplicate.

Reply:
oh, I see now, OK